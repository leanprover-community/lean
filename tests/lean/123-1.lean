/-- notation typeclass not in core.  -/
class has_scalar (G : Type) (V : Type) := (smul : G → V → V)

infixr ` • `:73 := has_scalar.smul

structure Ring : Type.

instance : has_coe_to_sort Ring :=
{
  S := Type,
  coe := λ R, unit
}

variables {G : Type} [has_mul G] {R : Ring}

class distrib_mul_action' (G : Type) (V : Type)
  [has_mul G]
  extends has_scalar G V :=
(foo : ∀ (x y : G) (b : V), x • b = x • y • b) -- note that changing x • y • b to y • b fixes the violation
                                               -- and instead we get a different error.

structure finsupp' (β : Type) : Type :=
(to_fun             : β → β)

def mas (r : R) : finsupp' ↥R  :=
{
  to_fun := id,
}

variables (V : Type)

instance foo : distrib_mul_action' G V :=
{
  smul := λ g v, (mas ()) • v, -- note that changing () to 37 also causes an assertion violation
  foo := λ g g' v, sorry
}
