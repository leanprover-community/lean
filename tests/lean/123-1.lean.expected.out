123-1.lean:34:18: error: don't know how to synthesize placeholder
context:
G : Type,
_inst_1 : has_mul G,
V : Type,
g : G,
v : V
‚ä¢ Ring
